tests/ha_cases/run.sh

```diff
@@ -470,60 +560,101 @@ function test_multi_task_reduce_and_restart_worker() {
+    isolate_master $follower_idx "isolate"
+    for idx in "${!alive[@]}"; do
+        if [[ ${alive[idx]} = $follower_idx ]]; then
+            unset 'alive[idx]'
+        fi
+    done
+    alive=("${alive[@]}")
 
-        run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$port" \
-            "pause-task test"\
-            "\"result\": true" 3
+    # find which worker is in use
+    task_name=(test test2)
+    worker_inuse=("")     # such as ("worker1" "worker4")
+    status=$($PWD/bin/dmctl.test DEVEL --master-addr "127.0.0.1:$new_leader_port" query-status test \
+        | grep 'worker' | awk -F: '{print $2}')
+    echo $status
+    for w in ${status[@]}; do
+        worker_inuse=(${worker_inuse[*]} ${w:0-9:7})
+        echo "find workers: ${w:0-9:7} for task: test"
+    done
+    echo "find all workers: ${worker_inuse[@]} (total: ${#worker_inuse[@]})"
 
-        run_sql "DROP TABLE if exists $ha_test.ta;" $TIDB_PORT $TIDB_PASSWORD
-        run_sql "DROP TABLE if exists $ha_test.tb;" $TIDB_PORT $TIDB_PASSWORD
-        load_data $MYSQL_PORT1 $MYSQL_PASSWORD1 "a" &
-        load_data $MYSQL_PORT2 $MYSQL_PASSWORD2 "b" &
+    for ((i=0; i < ${#worker_inuse[@]}; i++)); do
+        wk=${worker_inuse[$i]:0-1:1} # get worker id, such as ("1", "4")
+        echo "try to isolate dm-worker$wk"
+        isolate_worker $wk "isolate"
+    done
 
-        run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$port" \
-            "resume-task test"\
+    run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$new_leader_port" \
+            "pause-task test"\
             "\"result\": true" 3
 
-        # wait for load data
-        wait
-        sleep 3
-
-        echo "use sync_diff_inspector to check increment data"
-        check_sync_diff $WORK_DIR $cur/conf/diff_config.toml 10
+    run_sql "DROP TABLE if exists $ha_test.ta;" $TIDB_PORT $TIDB_PASSWORD
+    run_sql "DROP TABLE if exists $ha_test.tb;" $TIDB_PORT $TIDB_PASSWORD
+    load_data $MYSQL_PORT1 $MYSQL_PASSWORD1 "a" &
+    load_data $MYSQL_PORT2 $MYSQL_PASSWORD2 "b" &
 
-        isolate_master $idx "disable_isolate"
-        check_rpc_alive $cur/../bin/check_master_online 127.0.0.1:$port
-    done
+    run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$new_leader_port" \
+        "resume-task test"\
+        "\"result\": true" 3 # wait for load data wait
+    wait
+    sleep 3
 
+    echo "use sync_diff_inspector to check increment data"
+    check_sync_diff $WORK_DIR $cur/conf/diff_config.toml 10
 
-    echo "[$(date)] <<<<<< finish test_isolate_master >>>>>>"
+    echo "[$(date)] <<<<<< finish test_isolate_master_and_worker >>>>>>"
 }
 
 
 function run() {
-    test_join_masters_and_worker               # TICASE-928, 930, 931, 961
-    test_kill_master                           # TICASE-996
-    test_kill_worker                           # TICASE-968, 973, 1002
+    test_join_masters_and_worker               # TICASE-928, 930, 931, 961, 932, 957
+    test_kill_master                           # TICASE-996, 944, 958
```

Comment:
`test_kill_master` seems not _restart_ as `TICASE-944` expecteed.

Reply:
Is there any comments about `TICASE-945`?

Reply:
both 944 and 945 have been covered in `test_list_member`, 75c8977 add comments for them.